client/src/components/dataset/sidebars/ChangeDataType.jsx

```diff
@@ -6,32 +6,23 @@ import SidebarControls from './SidebarControls';
 import DateFormatSelect from './DateFormatSelect';
 
 function DefaultValueInput({ defaultValue, onChange, newType }) {
+  const emptyValue = newType === 'number' ? 0 : '';
```

Comment:
@peeb I think if you wanted to fix the "is not a NaN" console warning, you would have to add an explicit NaN check on line 17 (i.e. `(defaultValue !== null && !isNan(defaultValue) ? ...`